keps/sig-network/2112-netpol-namespace-name-selector/README.md

```diff
@@ -0,0 +1,371 @@
+need for developer driven security boundaries, these technologies aren't 
+available in most clusters, and aren't supported by the Kubernetes API.
+
+The ability to provide granular and intuitive network boundaries between apps 
+is part of the broader vision to make the NetworkPolicy API a universal 
+security construct implemented in all production applications.  Many 
+conversations have come up around this topic, with [kubernetes #88253](https://github.com/kubernetes/kubernetes/issues/88253) 
+being one of the most recent ones.
+
+### Generic motivation
+
+- Embrace the immutable nature of a namespace name as a fundamental security 
+construct in the Kubernetes ecosystem.
+- Making network policies more secure by making it impossible to "impersonate" 
+a bespoke namespace.
+- Making network policy tenancy boundaries more declarative to use by *not* 
+requiring developers to copy/duplicate namespace labels
+
+These three motivating factors are mostly self explanatory, but in the next 
+section we outline concrete feedback in these areas.
+
+### Community feedback on the namespace.Name selector
+
+- In 2016, the argument for namespace as name selectors was first made in
+the [mailing list](https://groups.google.com/g/kubernetes-sig-network/c/GzSGt-pxBYQ/m/Rbrxve-gGgAJ), based on the fact that labels can be 
+retroactively added to namespaces pretty easily, even if your namespace isn't 
+intended to be able to send traffic somewhere: 
+
+```
+we need to clarify how namespaces are matched. I'm pretty 
+uncomfortable with using labels here, since anyone can add labels at 
+any time to any namespace and thus send traffic to your pod if they 
+know the label you're using. If it were simply a namespace name, it's 
+much easier to specifically allow only namespaces you control since you 
+know your own namespace names. 
+```
+
+- Another recent argument was made on the basis of targeting namespaces 
+using names, rather then labels, on the basis of **sheer convenience**.
+
+```
+While matching things like pods etc by label is certainly worthwhile, when 
+matching a namespace I suspect the majority of the time you only want to match 
+a single namespace. It been great match against the name rather than just a 
+label. I suspect most people don't think to add labels to the namespaces.
+```
+
+The latter comment received 9 likes as a github issue - indiciating the 
+general popularity of this as a request, and that correlates well to feedback 
+we've seen in the broader community as well.
+
+We thus conclude that, even though targeting an object using its name is not 
+normal in K8s, in the case of NetworkPolicies, the overwhelming need for 
+universal, easy to define security boundaries, makes a strong case for 
+amending the API to support a "special" selector for namespace names that is 
+**independent** of labels.
+
+### Goals
+
+- Add a `namespaceSelectorAllow` option (which is additive to the current `matchLabels` selector).
```

Comment:
This name is pretty confusing IMO. First, I think the term `Allow` is redundant cause you only specify namespaces you are trying to allow. Second the term `Selector` implies this selects against labels (or most users will think this at least). Prefer `namespaces` or namespaceNames`.

Reply:
yeah i dont like this name either.  just wanted something other then `namespaceSelector` since that was taken. 

I'll go to `namespaceNames`

Reply:
fixed